src/os/filestore/HashIndex.cc

```diff
@@ -318,7 +318,7 @@ int HashIndex::split_dirs(const vector<string> &path, int target_level) {
     return r;
   }
 
-  if (must_split(info, target_level)) {
+  if (cct->_conf->filestore_auto_split_dirs && must_split(info, target_level)) {
```

Comment:
personally, i don't really like the way to address the performance issue by adding yet another knob. but if you really want to disable dir-splitting, probably could do so in `HashIndex::apply_layout_settings()`, instead of in the recursive call.

Reply:
Hi tchaikov,

If we disable it in HashIndex::apply_layout_settings(), we need to run ceph-objectstore-tool on every OSD. 

I understand your concern. How about we check filestore_auto_split_dirs option at the beginning of HashIndex::split_dirs()? Then there is no recursive call happened.